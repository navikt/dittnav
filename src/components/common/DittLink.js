import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Path } from '../../constants';

class DittLink extends Component {
  render() {
    const { url, children, className } = this.props;
    const onClick = (event) => {
      event.preventDefault();
      window.history.pushState(null, null, `${Path.CONTEXT}${url}`);
      window.dispatchEvent(new window.PopStateEvent('popstate'));
    };
    return (
      <a href={Path.CONTEXT} onClick={onClick} className={className}>
        { children }
      </a>
    );
  }
}

DittLink.propTypes = {
  url: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

DittLink.defaultProps = {
  className: '',
};

export default DittLink;
