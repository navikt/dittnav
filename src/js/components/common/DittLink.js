import React, { Component } from 'react';
import PropTypes from 'prop-types';
import conf from 'js/Config';

class DittLink extends Component {
  render() {
    const { url, children, className } = this.props;
    const onClick = (event) => {
      event.preventDefault();
      window.history.pushState(null, null, `${conf.dittNav.CONTEXT_PATH}${url}`);
      window.dispatchEvent(new window.PopStateEvent('popstate'));
    };
    return (
      <a href={conf.dittNav.CONTEXT_PATH} onClick={onClick} className={className}>
        { children }
      </a>
    );
  }
}

DittLink.propTypes = {
  url: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

DittLink.defaultProps = {
  className: '',
};

export default DittLink;
