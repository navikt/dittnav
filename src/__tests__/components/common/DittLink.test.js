import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import DittLink from 'components/common/DittLink'

test('DittLink renders', () => {
  const component = ReactTestRenderer.create(<DittLink url='nav.no'><div>hello</div></DittLink>);
  expect(component.toJSON()).toMatchSnapshot();
});
