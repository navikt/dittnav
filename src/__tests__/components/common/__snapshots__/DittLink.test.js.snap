// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DittLink renders 1`] = `
<a
  className=""
  href="/dittnav"
  onClick={[Function]}
>
  <div>
    hello
  </div>
</a>
`;
